torch/distributed/pipelining/PipelineSchedule.py

```diff
@@ -685,33 +682,29 @@ def __init__(
 
         self.n_local_stages = len(stages)
         self.rank = stages[0].group_rank
-
-    def _step_microbatches(
-        self,
-        arg_mbs: Optional[List] = None,
-        kwarg_mbs: Optional[List] = None,
-        target_mbs: Optional[List] = None,
-        losses: Optional[List] = None,
-    ):
-        """
-        Operate on the microbatches for interleaved 1f1b schedule (https://arxiv.org/pdf/2104.04473.pdf).
-
-        Highest rank has a warmup (fwd only) count of [len(stages) - 1] * number of PP ranks
-        and each rank away from highest rank adds 2 warmup steps due to:
-            - one happened before highest rank's warmup started,
-            - one waiting for backward result to trickle down from highest rank
-
-        TODO: Interleaved 1F1B does not support using _sorted_batch_p2p()
-        because it requires recvs and sends from different peers
-        to execute in the same coalesced operation. As a result, this schedule does
-        not support models with skip connections.
-        """
-        arg_mbs, kwarg_mbs = self._check_inputs(arg_mbs, kwarg_mbs, target_mbs, losses)
-
-        # increment warmup_steps by 2 for each hop away
-        warmup_steps = (self.n_local_stages - 1) * self.pp_group_size
-        warmup_steps += 2 * ((self.pp_group_size - 1) - self.rank)
-        warmup_steps = min(warmup_steps, self._n_microbatches * self.n_local_stages)
+        self.group = stages[0].group
+
+        # 1. Create the pipeline_order (all ranks do this calculation)
+        # This will be used to keep track of the current state of the entire pipeline
+        # pipeline_order[rank] = [(computation_type, microbatch_index, stage_index), ...]
+        self.pipeline_order: List[
+            List[Optional[Tuple[_ComputationType, int, int]]]
+        ] = []
```

Comment:
> pipeline_order[rank] = [(computation_type, microbatch_index, stage_index), ...]

For the left-hand side, use dict with `rank` as key, rather than using list.
For the right-hand side, document what the `List` structure is for.

Reply:
rank is local_rank so it will always be 0 to group_size. But I will update this and also update the tuple to a namedtuple.